prototype.tf

```diff
@@ -0,0 +1,346 @@
+resource "google_pubsub_subscription" "ingestion_subscription" {
+  name  = var.ingestion_subscription_name
+  topic = google_pubsub_topic.upload_to_gcs.name
+
+  ack_deadline_seconds = 60
+
+  push_config {
+    # Due to Terraform config language restrictions, index the first status element in a list of one.
+    push_endpoint = google_cloud_run_service.ingestion_service.status.0.url
+    oidc_token {
+      service_account_email = google_service_account.ingestion_invoker_identity.email
+    }
+  }
+}
+
+# Cloud Run service for uploading data to gcs.
+resource "google_cloud_run_service" "ingestion_service" {
+  name     = var.run_ingestion_service_name
+  location = var.compute_region
+  project  = var.project_id
+
+  template {
+    spec {
+      containers {
+        image = format("gcr.io/%s/%s", var.project_id, var.ingestion_image_name) 
+        env {
+          name  = "PROJECT_ID"
+          value = var.project_id
+        }
+        env {
+          name  = "NOTIFY_DATA_INGESTED_TOPIC"
+          value = var.notify_data_ingested_topic
+        }
+      }
+      service_account_name = google_service_account.ingestion_runner_identity.email
+    }
+  }
+
+  traffic {
+    percent         = 100
+    latest_revision = true
+  }
+  autogenerate_revision_name = true
+}
+
+# Push subscription for notify_data_ingested topic that invokes the run service.
+resource "google_pubsub_subscription" "notify_data_ingested_subscription" {
+  name  = var.notify_data_ingested_subscription_name
+  topic = google_pubsub_topic.notify_data_ingested.name
+
+  ack_deadline_seconds = 60
+
+  retry_policy {
+    minimum_backoff = "30s"
+  }
+
+  push_config {
+    # Due to Terraform config language restrictions, index the first status element in a list of one.
+    push_endpoint = google_cloud_run_service.gcs_to_bq_service.status.0.url
+    oidc_token {
+      service_account_email = google_service_account.gcs_to_bq_invoker_identity.email
+    }
+  }
+}
+
+# Cloud Run service for loading GCS buckets into Bigquery.
+resource "google_cloud_run_service" "gcs_to_bq_service" {
+  name     = var.run_gcs_to_bq_service_name
+  location = var.compute_region
+  project  = var.project_id
+
+  template {
+    spec {
+      containers {
+        image = format("gcr.io/%s/%s", var.project_id, var.gcs_to_bq_image_name)
+        env {
+          # Name of BQ dataset that we will add the tables to. This currently points to the main BQ dataset.
+          name  = "DATASET_NAME"
+          value = var.bq_dataset_name
+        }
+
+        resources {
+          limits = {
+            memory = "2G"
+          }
+        }
+      }
+      service_account_name = google_service_account.gcs_to_bq_runner_identity.email
+    }
+  }
+
+  traffic {
+    percent         = 100
+    latest_revision = true
+  }
+  autogenerate_revision_name = true
+}
+
+/* [END] Cloud Run Setup */
+
+
+/* TODO: replace scheduler setup with Airflow */
```

Comment:
nit: prefer to create an issue instead of leaving TODOs in code

Reply:
I filed an issue and linked to it from the TODO